pages/_app.js

```diff
@@ -121,6 +91,19 @@ export default function App({ Component, pageProps }) {
     return () => clearInterval(interval);
   }, [isActive3]);
 
+  useEffect(() => {
+    setTrys(trys);
+    if (checkFruits) {
+      tryText;
+      save();
+
+      setTrys(1);
+    }
+    if (!checkFruits && !isActive3 && !isActive1 && !isActive2) {
+      setTrys(trys + 1);
+    }
+  }, [isActive3 || isActive2 || isActive1]);
```

Comment:
missing dependecies:
[checkFruits, isActive1, isActive2, isActive3, save, tryText, trys]);

Reply:
problem: you are calling setFunction inside useEffect. this causes infinite loop